### PR TITLE
community: Add async methods to the AstraDB BaseStore

### PR DESCRIPTION
**Pull request by cbornet**

### CODE REVIEW

**Comment by cbornet** [comment]
Ready for review.
cc @hemidactylus

**Review by baskaryan — APPROVED** [review]
looks like there's a merge conflict (sorry for slow review), otherwise lgtm!

File: libs/community/langchain_community/storage/astradb.py [hunk 1]
```diff
@@ -30,21 +34,27 @@ class AstraDBBaseStore(Generic[V], BaseStore[str, V], ABC):
...
         self,
         collection_name: str,
+        *,
         token: Optional[str] = None,
         api_endpoint: Optional[str] = None,
         astra_db_client: Optional[AstraDB] = None,
+        async_astra_db_client: Optional[AsyncAstraDB] = None,
         namespace: Optional[str] = None,
+        pre_delete_collection: bool = False,
+        setup_mode: SetupMode = SetupMode.SYNC,
```
Comment (baskaryan) [inline libs/community/langchain_community/storage/astradb.py:44]: this is breaking, would prefer
```suggestion
    def __init__(
        self,
        collection_name: str,
        token: Optional[str] = None,
        api_endpoint: Optional[str] = None,
        astra_db_client: Optional[AstraDB] = None,
        namespace: Optional[str] = None,
        *,
        async_astra_db_client: Optional[AsyncAstraDB] = None,
        pre_delete_collection: bool = False,
        setup_mode: SetupMode = SetupMode.SYNC,
```

Comment (cbornet) [reply libs/community/langchain_community/storage/astradb.py]: Sure 👍

**Comment by cbornet** [comment]
BTW tag is wrong. It shouldn't be `chroma` but `astradb`

**Pull request merged**